core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -360,7 +360,7 @@ class LogManager(logDirs: Seq[File],
       for ((cleanShutdownFile, dirJobs) <- jobs) {
         dirJobs.foreach(_.get)
         try {
-          cleanShutdownFile.delete()
+          Files.deleteIfExists(cleanShutdownFile.toPath)
```

Comment:
Just checking, but the issue here is that we might mistakenly mark the directory is offline if the clean shutdown file did not exist?

Reply:
`File.delete` doesn't throw an exception and we don't check the result. So the previous code was very misleading. That's what I was trying to fix. And to avoid introducing an issue if the file did not exist, I am using `deleteIfExists`.